lib/plek.rb

```diff
@@ -37,8 +34,14 @@ class NoConfigurationError < StandardError; end
   #   +GOVUK_APP_DOMAIN_EXTERNAL+ and if that is unavailable the parent domain
   #   will be used
   def initialize(domain_to_use = nil, external_domain = nil)
-    self.parent_domain = domain_to_use || env_var_or_dev_fallback("GOVUK_APP_DOMAIN", DEV_DOMAIN)
-    self.external_domain = external_domain || ENV["GOVUK_APP_DOMAIN_EXTERNAL"] || parent_domain
+    truth_re = /^[1tTyY]/
+    @parent_domain = domain_to_use || env_var_or_dev_fallback("GOVUK_APP_DOMAIN", DEV_DOMAIN)
+    @external_domain = external_domain || ENV.fetch("GOVUK_APP_DOMAIN_EXTERNAL", @parent_domain)
+    @host_prefix = ENV.fetch("PLEK_HOSTNAME_PREFIX", "")
+    # TODO: switch to a data structure with with constant-time lookups if we
+    # end up with more than a tiny number of unprefixable_hosts entries.
+    @unprefixable_hosts = ENV.fetch("PLEK_UNPREFIXABLE_HOSTS", "").split(",").map(&:strip)
```

Comment:
Did you want to run uniq on this? I guess it's not a problem if it has duplicate entries?

Reply:
I thought about it but it seems like a waste of cycles and doesn't help readability, so gave it a miss.